spec.emu

```diff
@@ -15,37 +15,68 @@ contributors: Jordan Harband
 	<emu-alg>
 		1. Let _promise_ be the *this* value.
 		1. If IsPromise(_promise_) is *false*, throw a *TypeError* exception.
-		1. Let _thenFinally_ be ! CreateThenFinally(_onFinally_).
-		1. Let _catchFinally_ be ! CreateCatchFinally(_onFinally_).
+		1. Let _C_ be ? SpeciesConstructor(_promise_, %Promise%).
+		1. Let _thenFinally_ be ! CreateThenFinally(_C_, _onFinally_).
+		1. Let _catchFinally_ be ! CreateCatchFinally(_C_, _onFinally_).
 		1. Return ? Invoke(_promise_, `"then"`, &laquo; _thenFinally_, _catchFinally_ &raquo;).
 	</emu-alg>
 </emu-clause>
 
 <emu-clause id="sec-createthenfinally" aoid="CreateThenFinally">
-	<h1>CreateThenFinally ( _onFinally_ )</h1>
-	<p>The abstract operation CreateThenFinally takes an _onFinally_ function, and returns a callback function for use in Promise.prototype.finally.</p>
+	<h1>CreateThenFinally ( _C_, _onFinally_ )</h1>
```

Comment:
You'll want to wrap these in emu-clauses for nesting. I think the most-consistent would be

1. Promise.prototype.finally
   1. CreateThenFinally
      1. ThenFinally Functions
   1. CreateCatchFinally
      1. CatchFinally Functions

although flattening those to one level under Promise.prototype.finally also seems OK to me. (Or, as suggested above, inline it, so there are only two sub-clauses.)